odoo/addons/base/models/avatar_mixin.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from hashlib import sha512
+from base64 import b64encode
+
+from odoo import models, fields, api
+
+from odoo.tools import html_escape, file_open
+from odoo.modules.module import get_resource_path
+
+
+class AvatarMixin(models.AbstractModel):
+    _name = 'avatar.mixin'
+    _inherit = ['image.mixin']
+    _description = "Avatar Mixin"
+
+    # all image fields are base64 encoded and PIL-supported
+
+    avatar_1920 = fields.Image("Avatar", max_width=1920, max_height=1920, compute="_compute_avatar_1920", readonly=False)
+
+    avatar_1024 = fields.Image("Avatar 1024", max_width=1024, max_height=1024, compute="_compute_avatar_1024", readonly=False)
+    avatar_512 = fields.Image("Avatar 512", max_width=512, max_height=512, compute="_compute_avatar_512", readonly=False)
+    avatar_256 = fields.Image("Avatar 256", max_width=256, max_height=256, compute="_compute_avatar_256", readonly=False)
+    avatar_128 = fields.Image("Avatar 128", max_width=128, max_height=128, compute="_compute_avatar_128", readonly=False)
+
+    def _compute_avatar(self, avatar_field, image_field):
+        for record in self:
+            record[avatar_field] = record[image_field] or record._get_one_avatar(avatar_field, image_field)
+
+    def _get_one_avatar(self, avatar_field, image_field):
```

Comment:
Could you make a real batch method ? We should avoid "one" method (both in naming and work), I bet you can do everything in batch using record sets, this is quite easy with filtering.

Reply:
The goal of this method is to allow easy override (partner & employee to look into related user), it's actually quite complex to cleanly override a batch method for specific records without repeating the whole logic in the override.

Reply:
Unless I am mistaken it would look like ``super(self - done)._avatar_compute_avatar() `` ? Well maybe having a single method could ease inheritance if you have a lot, but that the way we always do in python, work with record sets and split it into sub-record sets.

Reply:
This might actually work here, @leonardopavanrocha could you have a look? Let me know if you need help

Reply:
This was done, thanks for the suggestion.

Any final word from Mixin Master?